src/dtsh/cli.py

```diff
@@ -98,6 +119,24 @@ def theme(self) -> Optional[str]:
         """Additional styles file."""
         return str(self._argv.theme[0]) if self._argv.theme else None
 
+    @property
+    def batch_source(self) -> Optional[Union[str, List[str]]]:
+        """Batch command source, if defined."""
+        if self._argv.c:
+            return cast(List[str], self._argv.c)
+        elif self._argv.f:
+            return self._argv.f
+        else:
+            return None
+
+    @property
+    def interactive(self) -> bool:
+        """Is the interactive loop requested?"""
+        if not self._argv.c and not self._argv.f:
+            # no batch input, must be interactive
+            return True
+        return self._argv.interactive
```

Comment:
Type hinting again, sorry for this.

```suggestion
        if not (self._argv.c or self._argv.f):
            # no batch input, must be interactive
            return True
        return cast(bool, self._argv.interactive)
```